http/src/main/java/org/http4s/blaze/http/parser/BodyAndHeaderParser.java

```diff
@@ -386,6 +387,7 @@ else if (ch == HttpTokens.LF) {
                     final int buff_size = in.remaining();
 
                     if (remaining_chunk_size <= buff_size) {
+                        // End of this chunk
```

Comment:
I don't think it needs to be a part of this PR, hence why I stated it was tangential. It just came to my attention.

The docs were just something that when trying to analyze a state machine clear documentation is a must. Otherwise it's almost an attempt to climb into another developers mind.

The benefits I seeing moving to Scala is considering the interesting developments with ScalaJS and ScalaNative. Which are far away from being able to be integrated with blaze, but may be interesting if ScalaNative makes progress. So if we were capable of shifting to pure Scala, that could be a beneficial move.